% !TeX document-id = {a8f37c7e-a6b2-4962-8b1d-d64617ca6a40}
% !TEX TS-program = LaTex
% !TEX encoding = UTF-8 Unicode
% !BIB TS-program = biber
% !BIB program = biber

\documentclass[11pt]{scrartcl}
\usepackage{geometry} % See geometry.pdf to learn the layout options. There are lots.
\geometry{a4paper}  % ... or a4paper or a5paper or ... 
%\geometry{landscape} % Activate for for rotated page geometry
%\usepackage[parfill]{parskip} % Activate to begin paragraphs with an empty line rather than an indent
\usepackage{graphicx}
\graphicspath{{./media/}}
\usepackage{amssymb}
\usepackage[toc,page]{appendix}
%\usepackage[pagewise,modulo]{lineno}
\usepackage{enumitem}
\usepackage{minted}
\newminted[bash]{bash}{breaklines, breaksymbol={},breakindent=1cm}
\newmintinline{bash}{}
\usepackage{hyperref}
\usepackage{nameref}
\usepackage[en-GB]{datetime2}
\usepackage{subfigure}
% using biber!!
\usepackage[natbib=true,style=authoryear]{biblatex}
\addbibresource{alea.bib}
\usepackage{framed}
\usepackage{pifont}
\usepackage{comment}

\title{\textit{LIONS} Manual}
\subtitle{v.0.2}
\author{Artem Babaian, Richard Thompson}

\newcommand{\arrows}[1]{\textless #1\textgreater}

\begin{document}

\maketitle

\tableofcontents

%\linenumbers

\section{LIONS Pipeline Architecture}
LIONS is a bioinformatic analysis pipeline which brings together a
 few pieces of software and some home-brewed scripts to annotate a
 paired-end RNAseq library against a reference TE annotation set
 (such as Repeat Masker).

 \textbf{East Lion} processes a bam file input, re-aligns it to a genome,
 builds an ab initio assembly using Tophat2. This assembly is then
 proccessed and local read searches are done at the 5' ends to find
 additional transcript start sites and quality control the 5' ends of
 the assembly. The output is a file-type \arrows{library} .lions which
 annotates the intersection between the assembly, a reference gene set
 and repeat set.

 \textbf{West Lion} compiles different .lions files, groups them into biological
 catagories (i.e. Cancer vs. Normal or Treatment vs. Control) and
 compares and analyzes the data to create graphs and meaningful
 interpretation of the data.

\section{LIONS Installation}

Download/clone the LIONS repo: https://github.com/ababaian/LIONS/archive/master.zip

\subsection{LIONS Container (Docker)}

LIONS can be installed as a Docker container. Navigate to the \$LIONS folder which contains the `Dockerfile`. Build container with:

\begin{bash}
cd \$LIONS/
docker build -t lions .
\end{bash}

Users will still need to download the LIONS resource files (See below); although users wishing to use hg19 or hg38 can use the \bashinline{source_refs.sh} to automate this process.

In order to assist users, LIONS provides a reference-read 'Fastq to Output' Dockerfile, the resulting container should not be used for anaylsis due it's size but will allow users to understand LIONS usage and the resource directory layout. 
The example container can be built from the \$LIONS directory using:

\begin{bash}
docker build -t lions-ref Ref-Ready/.
\end{bash}
Please note, the example container does require the LIONS container to have already been built and named as described above.

\vspace{2em}
\subsection{LIONS Command Line}

Users with experience of the linux commandline may wish to download the package from github and run it directly without using Docker. This is especially useful for cluster-computing.In this case ensure the following software is installed on your system: 

\begin{itemize}
\item Python3 and pysam
\item Bowtie2
\item Tophat2
\item Java v8 or higher
\item Samtools v0.1.18
\item R v3.5.0 or higher
\item Bedtools v2.25.0
\item Cufflinks v2.2.1
\end{itemize}

\subsection{Initializing LIONS Resources}

\begin{enumerate}
\item If you don't have LIONS downloaded already; clone it
\begin{bash}
git clone git@github.com:ababaian/LIONS.git
\end{bash}

\item Initialize genomic resources for LIONS in./LIONS/resources copy 'example' folder to '\arrows{genomeName}' folder

\item Populate the resource files:
\begin{enumerate}
\item In ./LIONS/resources/\arrows{genomeName}/genome/ add a \arrows{genomeName}.fa genome sequence file\footnote{2bit files can be converted to fa using twoBitToFa available from UCSC (\url{http://hgdownload.cse.ucsc.edu/admin/exe/linux.x86_64/twoBitToFa}).}
\item In ./LIONS/resources/\arrows{genomeName}/annotation/ add the ucsc annotation file
\item In ./LIONS/resources/\arrows{genomeName}/repeat/ add the ucsc repeatMasker file

\begin{bash}
# genomeName = hg38 ========================
# Genome
http://hgdownload.soe.ucsc.edu/goldenPath/hg38/bigZips/hg38.fa.gz

# Gene Annotation (RefSeq)
https://s3-us-west-2.amazonaws.com/lionproject/resources\
/hg38/refseq_hg38.ucsc.gz

# Repeat Masker
https://s3-us-west-2.amazonaws.com/lionproject/resources\
/hg38/rm_hg38.ucsc.gz
	
# genomeName = hg19 ========================
# Genome
http://hgdownload.soe.ucsc.edu/goldenPath/hg19/bigZips/hg19.2bit

# Gene Annotation (RefSeq)
https://s3-us-west-2.amazonaws.com/lionproject/resources\
/hg19/refSeq_hg19.ucsc.zip

# Repeat Masker
https://s3-us-west-2.amazonaws.com/lionproject/resources\
/hg19/rm_hg19.ucsc.zip

\end{bash}

\textit{(The UCSC files are downloaded from the UCSC Table Browser as the `all fields from selected table` output format)}
\end{enumerate}


\item Initialize the project parameters in parameter.ctrl:
\begin{enumerate}
\item Give your project a name and fill out all the parameters
 software designated in the software list.
\end{enumerate}


\item This file contains all the \textit{system-specific} parameters for LIONS to run on your computer. Go through this file and ensure the parameters make sense and the software pointers are compatible.
\begin{enumerate}
\item Give your project a name and fill out all the parameters as you see fit.
\item In the software list, refer LIONS to the paths for each software on your system. If it's in \$BIN already then you can simply type the command; LIONS will create links in it's own folder structure to the software designated in the software list.
\end{enumerate}
\end{enumerate}

\clearpage




\section{Running LIONS}
\label{lions_run}

Once the container has been built, lions can be run inside an interactive docker container
which can be started using the following command:

\begin{bash}
docker run -ti lions
\end{bash}

It's a good idea to create a local LIONS directory (\$LIONS) outside of the container with all the resources and parameter files set-up. You can then mount the \$LIONS directory into the container using \textit{-v}.

This command will open a (bash) terminal to the container, in the LIONS base directory.
 
\begin{bash}
cd \$LIONS
docker run -v \$LIONS:/LIONS -ti lions
\end{bash}

where `\$LIONS` is the directory on the host machine containg the data for analysis and `/LIONS` is the directory inside the container where the data will be accessed.

\textit{e.g.} \bashinline{Docker run -v /home/artem/LIONS-master:/LIONS -ti lions} will allow the user to access to the contents of /home/artem/LIONS-master within the container.

\begin{framed}
Alternatively you can load the resource files and data files into container at start-up.

\begin{bash}
-v <resource_directory>:/LIONS-docker/resources/<genomeName> \
-v <data_directory>:/LIONS-data/
\end{bash}
for example:
\begin{bash}
-v ~/hg19:/LIONS-docker/resources/hg19 \
-v ~/ENCODE_bams/:/LIONS-data
\end{bash}

this will avoid adding resources to the image eachtime the container is run, or swelling the container size.
\end{framed}

\subsection{Running LIONS using a parameter file}

The default parameter file for LIONS is \texttt{/LIONS/controls/parameter.ctrl}. 

However, the user can run LIONS with with a specified parameter file as follows:
\begin{bash}
bash lions.sh <path/to/parameter/ctrl>
\end{bash}

\textit{\textbf{N.B.}} As the docker container will not carry across changes to the file structure, such as saved output files, it is advised that the mount point identified with the \texttt{-v} option is used to save output files and modified \texttt{parameter.ctrl} and \texttt{input.list} files, in addition to supplying input files.

\newpage
\subsection{Running LIONS using commandline options}

Lions can be run andline th commandline options using the \bashinline{lions_opt.sh} script. Any options not provided on the commandline will be taken from the default parameter file.
\begin{bash}
./lions_opt.sh [options]
\end{bash}

Where options include:

\begin{description}
\item[-h | -{}-help] Show commandlines usage
\item[-P | -{}-parameter] Define user parameter.ctrl file, see section \ref{controls}; \textit{\textbf{N.B.}} This option must be defined first in order that the other options are used correctly. 
\item[-b | -{}-base \arrows{path}] Sets the LIONS base directory to be <path>, defaults to \$PWD.
\item[-p | -{}-project \arrows{project-name}] Set name of the project, and therefore the output directory name in the projects folder, to <project\_name>
\item[-i | -{}-inputlist \arrows{path/to/input.list}] Identifies the input file containing the sample names and input files for the analysis, see section \ref{controls}
%	# <libPath> is one of
%	#     A) Paired end bam file. '/home/libPath.bam'
%	#     B) Sorted paired fastQ files. Comma seperated.
%	#        '/home/lib.fq1,/home/lib.fq2'
\item[-{}-callsettings \arrows{pre-set-name}] Tells LIONS which settings to use for the TE-initiation. Expected values include: 
        \begin{description}
        \item['oncoexapatation'] Detect high abundance TE isoforms
        \item['transcriptomeANN'] Artifical Neural Network based classifier
        \item['screenTE'] High sensitivity, low specificity detection
        \item['driverTE'] TE-initiations as main drivers of gene-expression
        \item['custom'] Use custom settings defined below
        \end{description}
	default setting is \textbf{'oncoexaptation'}.

\item[-I | -{}-index \arrows{INDEX}] Sets the LIONS Resource index, see section \ref{resources}.
\item[-{}-geneset \arrows{geneset.file}] Options identifies a ucsc formatted file containing a gene set to be
  used in the analysis, see section \ref{resources}.
\item[-{}-repeatmasker \arrows{repeat.file}] Identifies a ucsc formatted file containing the repeatMasker
  annotation for the genome, see section \ref{resources}.

\item[-{}-systemctrl \arrows{system.ctrl}] Identifies the control file used for passing system settings to LIONS, see section \ref{controls} 
 
\item[-{}-bowtie \arrows{?}] ?Additional Bowtie options? .. or command?

\item[-{}-alignbypass \#] binary switch to tell LIONS whether to run a new alignment on input files, or to use that provided in the input files. Default is `\texttt{1}'.
       \begin{description}
	\item[0] Calculate a new alignment for the input
	\item[1] Do not re-calculate the alignment, simply create symbolic link to the bam file in \arrows{input.list} within the LIONS folder architecture
       \end{description}

\item[-{}-inread \#]  The Inner Read distance to be used by Bowtie during alignment (Bowtie's \texttt{-r} option). 

\item[-{}-threads \#] The number of parallel threads to be used by Tophat/Bowtie and Cufflinks during the analysis. \textbf{\textit{N.B.}} presently it is necessary to ensure that the \texttt{-{}-threads} option is passed after the \texttt{-{}-systemctrl}, if both are used, in order to ensure the threads option is used correctly throughout the analysis. This will be corrected in future versions.

% Cufflinks
\item[-{}-denovo \#] Tell Cufflinks whether to carry out a \textit{de novo} or \textit{ab initio} assembly, LIONS defaults to \textit{de novo}.
	\begin{description}
        \item[0] \textit{Ab Initio}
        \item[1] \textit{De Novo}
        \end{description}

\item[-{}-minfrags \#] Set Cufflink's `minimum fragments per transfrag', default is 10.
\item[-{}-multiflag \#] Set Cufflink's `multi-mapping transfrag fragments' option, default is 0.75.
\item[-{}-mintrim \#] Set Cufflink's `minimum coverage to attempt 3` trimming' option, default is 10.
\item[-{}-trimdrop \#] Set Cufflink's `minimum Trim dropoff in fraction' option.
\item[-{}-merger \#] Set Cufflink's `merge radius in bp', default is 50 bp.
	

%# RNAseqPipeline

\item[-{}-quality \arrows{qual}] RNAseq Analysis Quality, defined by the string `q\#.F\#' where  q = quality cutoff; F bam flags to discard, \textit{e.g.} `q10.F772' and `q1.F1796'. 


\end{description}
\begin{comment}


% ChimSort

	# Sort Bypass: If a sorted .lions file is already present
	# do you want to re-calcualate the 'initiations'?
	# 0 = Re-calcualte 'initiation' using parameters set below
	# 1 = Don't re-calculate 'initiation' if it exists
	export SORTBYPASS='0'

	# These parameters define what is defined as a 
	# "TE-initiated transcript" and what is exonization/termination are
	# note: See chimSort.R script to see how these are used for sorting
        # in more detail

if [ $CALLSETTINGS == 'custom' ]
then
	# Define call settings manually

	# Number of Chimeric Reads Required (total)
	export crtReads='3' # Variable: <Value below> or 1/20 RPM
	
	# Thread Ratio
	export crtThread='10' # >=
	export crtDownThread='10' # |number| required 

	# RPKM cut-off to consider an exon 'expressed'
	export crtRPKM='1' # >=

	# Contribution of TE to total transcript expression (Exon / TE)
	export crtContribution='0.1' # >=

	# Expression immediatly upstream TE
	export crtUpstreamCover='2' # >=

	# Expression of exons upstream of involved exon
	export crtUpstreamExonCover='1.5' # >=

	# Splice Partner Classification
	# Repeat Rank > 0 AND RepeatExonic
	
	# Final Custom Export String
	export CRT=$( echo $crtReads $crtThread $crtDownThread $crtRPKM $crtContribution $crtUpstreamCover $crtUpstreamExonCover )



# WEST LION Parameters-{}--{}--{}--{}--{}--{}--{}--{}--{}--{}--{}--{}--{}--{}--{}--{}--{}--{}--{}--{}--{}--{}--{}--
# chimGroup.R
	# These parameters define recurrent and specific
	# 'TE initiated' events
	# 
	# The third column of <input.list> defines libraries as one of
	# 1: Control / Normal
	# 2: Experimental / Cancer
	# 3: Other / unused

	# Recurrent Cutoff
	# Must be present in at least this many Group 2 libraries
	export cgGroupRecurrence='1'

	# Specificity Cutoff
	# Must be absent in at least this many Group 1 libraires
	export cgSpecificity='1'

	# chimGroup command string
	export CG=$(echo $cgGroupRecurrence $cgSpecificity)
\end{comment}

\clearpage
\section{LIONS File Architecture}

% ===================================================================
\subsection{./}
  base directory:  LIONS is a self-contained pipeline and 
  references needed by LIONS from the system are linked
  within this directory.

\begin{description}
\item[./lions.sh \arrows{parameter.ctrl}]
    The master script from which the entire pipeline is ran
    The script reads and processes all files in input.list
    and all parameters can be controlled from parameter.ctrl
\end{description}

\subsection{./controls}
\label{controls}
  Control Files: This folder contains project and system-specific
  parameters for running LIONS. There are three main files which
  need to be set-up, LIONS run parameters, system parameters and
  input RNA-seq libraries.
  
\begin{description}
\item[./controls/parameter.ctrl]
    A bash script which defines global project-specific 
    variables such as Project Name, library input list etc...
    The .sysctrl and .list file are defined here as well

\item[./controls/system.sysctrl]
    A bash script which defines global variables for all LIONS
    scripts. Also defines system-specific variables such as
    System Name, number of CPU cores etc...

\item[./controls/input.list]
    A three column tab-delimited file defining:

    \arrows{Library\_Name}\hspace{2em}\arrows{LibPath}\hspace{2em}\arrows{Grouping}

where \texttt{Grouping} is one of:
\begin{center}
    \begin{description}
    \item[1] control
    \item[2] experimental
    \item[3] other
    \end{description}
\end{center}
 .. and \texttt{<libPath>} is either a paired end bam file, \textit{e.g.} `/home/libPath.bam', or a pair of sorted fastQ files (as a comma seperated list), \textit{e.g.} `/home/lib.fq1,/home/lib.fq2'. Please note that each line should have a unique Library name.
\end{description}
    
\subsection{./bin}
  LIONS internal folder for symbolic links to binaries needed by
  the pipeline and script to initialize the folder. Make sure to set
  the correct commands for the software list in parameters.ctrl for
  your system


\subsection{./projects}
  For each \arrows{Project\_Name} a single folder will be initilized in which
  the data will be organized.

\begin{description}
\item[./projects/\arrows{Project\_Name}]
  The main directory for this project. Each individual library in the
  input will have a folder generated here called \arrows{Library}.

\item[./projects/\arrows{Project\_Name}/logs]
  Folder contains run-specific information such as input file at time
  of run and a copy of the input parameter file at time of run.

\item[./projects/\arrows{Project\_Name}/Analysis(\_RUNID)]
  Not implemented yet. This contains all data analysis for a run
  of LIONS. All graphs and project-wide .lions files are stored here
  
\item[./projects/\arrows{Project\_Name}/\arrows{Library}]
  Library-specific data and primary analysis files.

\item[./projects/\arrows{Project\_Name}/\arrows{Library}/\arrows{Library}.lcsv]
  Raw output file from LIONS containing all possible TE-Exon interaction
  data. This will include initiations, exonizations and terminations
  along with many calculated values about these loci from which
  LIONS will sort initiation events from the others. \arrows{Library}.pc.lcsv
  is the same file with additional information about overlapping protein
  coding genes.

\item[./projects/\arrows{Project\_Name}/\arrows{Library}/\arrows{Library}.lions]
  Initiation only TE-exon data from post-sorting. This file is the 
  complete list of transcripts initiated by TEs in this library. This
  data is passed on to the West Lion protocol to compare TE usage
  between libraries.

\item[./projects/\arrows{Project\_Name}/\arrows{Library}/alignment]
  tophat2-generated alignment and the re-aligned .bam file which
  will be used for analysis. Also contains flagstats and log files.
  Note: Once the alignment is generated it will not be re-generated
  even if you change the alignment parameters in parameter.ctrl. To
  re-make alignments simply start the project with a new name or delete
  these files.

\item[./projects/\arrows{Project\_Name}/\arrows{Library}/assembly]
  cufflinks-generated assembly in 'transcripts.gtf'

\item[./projects/\arrows{Project\_Name}/\arrows{Library}/expression]
  The output from a series of custom scripts 'RNAseqPipeline' which
  will generate wig files and perform RPKM calculations on a series 

\item[./projects/\arrows{Project\_Name}/\arrows{Library}/resources]
  Charlie-foxtrot of library-specific files used to calculate a score
  of parameters in the pipeline.
\end{description}

 \subsection{./scripts}
  Scripts: all scripts to run lions are held here except for the
  controlling lions.sh script which is in the base folder.
  Check initializeScripts.sh for complete list of scripts
  The main scripts are 

\begin{description}
\item[./scripts/eastLion.sh]
    Alignment, Assembly, Chimeric Detection pipeline

\item[./scripts/westLion.sh]
    LIONS analysis pipeline
\end{description}

  \subsection{./resources}
  \label{resources}
  Input files containg resource information needed for LIONS to run the analysis. The geneset and RepeatMasker files should be set in the \texttt{parameter.ctrl} file.
  
\begin{description}
\item[./resources/\arrows{INDEX}/annotation/\arrows{GENESET}] A ucsc formatted file containing a gene set to be
  used in the analysis (i.e. look for overlapping genes to transcripts)
  Download from: https://genome.ucsc.edu/cgi-bin/hgTables
  The standard annotation set used was RefSeq 'RefGene' table.

\item[./resources/\arrows{INDEX}/genome/\arrows{INDEX}.fa] The only requisite file here for running LIONS is a fasta
  formatted genome. This could be a symbolic link. LIONS will generate
  the other files necessary from INDEX.fa. If you have the .bt2 index
  files already generated you can symbolically link them in this folder
  to skip re-generating them.

\item[./resources/\arrows{INDEX}/repeat/\arrows{REPEATMASKER}.ucsc] a ucsc formatted file containing the repeatMasker
  annotation for the genome.
  (Download from UCSC genome browswer or format from RepeatMasker)
  Columns are;
  bin, swScore, milliDiv, milliDel, milliIns, genoName, genoStart,
  genoEnd, genoEnd, genoLeft, strand, repName, repClass, repFamily,
  repStart, repEnd, repLeft, id
\end{description}
\begin{framed}
\arrows{INDEX} : the name of the index set. To be compatible with different
  genome versions, species and gene sets there can be different sets
  of data. The \arrows{INDEX} global variable is set in parameter.ctrl file.
\end{framed}

  \subsection{./software}
  Packaged with LIONS is a few bits of software which will set-up your
  system to run the pipeline. Namely setuptools and pysam are the most
  challenging things to install. I found that it's easiest to set-up
  the pipeline using pip and download the package pysam from there.
  Pysam is used to read teh bam files in the python scripts.




\clearpage
\section{LIONS Error Codes}

% LIONS ERROR CODES

\begin{description}
\item[Error 1] Internal software error - Check last-run software.
\end{description}

\subsection{Initialization Codes}
\begin{description}


\item[Error 2] Initialization file missing or inaccesible -
  A file is missing or is unreadable. Ensure you have a complete version of LIONS and/or
make the missing script readable/exectable

\item[Error 3] A LIONS script is missing -
  A script is missing from \texttt{./LIONS/scripts/}; ensure your copy of LIONS is
complete or redownload.

\item[Error 4] Initialization bin missing -
  A binary is not found on the system. Configure \texttt{./LIONS/bin/initializeBin.sh} for your system

\item[Error 5] A resource file is missing or unreadable - 
  Checking/initialization of \texttt{./LIONS/scripts}.

\item[Error 6] A Python requisite is missing.

\item[Error 7] The input read file (.bam or .fastq) is non-readable or empty
\begin{description}
\item[7A] Bam file error
\item[7B] FastQ file error. Ensure the two files are comma seperated in the input
\end{description}
\end{description}

\subsection{eastLion Error Codes}
%% eastLion Codes -{}--{}--{}--{}--{}--{}--{}--{}--{}--{}--{}--{}--{}--{}--{}--{}-

\begin{description}
\item[Error 10] alignment not generated -
  An attempt was made to generate an alignment but the output file was
empty at the end of the script

\item[Error 12] wig not generated -
  An attempt was made to generate the wig file but the output file wasn't
present after the script ran
\end{description}

\subsection{westLion Error Codes}

%% westLion Codes -{}--{}--{}--{}--{}--{}--{}--{}--{}--{}--{}--{}--{}--{}--{}--{}--{}--
\begin{description}
\item[Error 15] A lions file wasn't generated -
  In the run, one of the lions files wasn't generated which means there
was an error. Don't run West Lions pipeline.
\end{description}





\clearpage
\section{LIONS output definitions}

\subsection{Output File Types}

LIONS produces several outputs from different stages of the analysis in addition to the standard outputs one would expect (.bam / .gtf).

\begin{description}
\item[`\arrows{library}.lcsv' / `.pc.lcsv']
  These are LIONS CSV files which contain the raw calculations for all major
  numeric operations.
  This includes ALL TE-exon interactions types (Initiation, Exonization and
  Termination). As such there are usually hundreds of thousands of TEs which
  have read fragments joining them to some assembled exon.

  The `.pc.' pre-suffix means the data has been intersected to the input
  set of protein coding genes.

  Use this file for re-calculating ``TE-Initiations'' with new parameters.

\item[`\arrows{library}.lion']
  This is the filtered set of TE-exon interactions which have been classified
  as ``TE-Initiations'' or TE transcription start sites. This is per-library
  input.

\item[`\arrows{project}.lions'] A merged file of several `.lion' files combining biological groups defined in the `input.list'. A good example of this is merging 10 cancer libraries and 10 normal libraries and outputing only those TE-initiations which are in at least 20\% of Cancer and no Normal libraries. These parameters can be changed in the `paramter.ctrl' input.

\item[`\arrows{project}.rslions'] The `rs' is for Recurrent and Specific TE-initiations only. That is if you   compare the set of libraries 1 (Normal) vs set 2 (Cancer), this contains only those TE-initiations which occur multiple times in Cancer (recurrant) and do not occur in Normal (specific). As defined by `\$cgGroupRecurrence' and `\$cgSpecificity' in the `paramter.ctrl' file.

\item[`\arrows{project}.inv.rslions'] The `.inv.' pre-suffix is simply the \textbf{inverse} of the `.rslion' file. So instead of ``Cancer vs. Normal'', ``Normal vs. Cancer''. A necessary control if one makes any conclusions based on enrichment/depletion.
\end{description}

\subsection{Output Columns}

\subsubsection{`\arrows{project}.lions'}
Most columns should be self-explanatory, some are not.

\begin{description}
\item[transcriptID] Unique identifier for the transcript (isoform). Usually taken
  from the assembly/reference transcriptome

\item[exonRankInTranscript] For each TE-exon interaction combination (row) which exon
  in the `transcriptID' is this row referring to

\item[repeatName] The \arrows{repeat\_name}:\arrows{repeat\_class}:\arrows{repeat\_family} taken from input
  set

\item[coordinates] Useful coordinates for visualizing the interaction. It starts/ends
  in the exon and repeat so when opening in a visualization tool you can see
  the reads spanning this area.

ER\_Interaction: The type of relative intersection in the genome between the exon
  and the repeat. Definitions are relative to the exon. Can be ``Up'', ``UpEdge'', ``EInside'', ``RInside'', ``Down'', ``DownEdge''.

\item[IsExonic] ??

\item[ExonsOverlappingWithRepeat] A list of \arrows{transcriptID:exonRank} which overlap
  the repeat.

\item[ER / DR / DE / DD / Total] A count of the number of TE-Exon sequence fragments
  which join this rows TE and Exon. ER means that one end overlaps the Exon
  and one end overlaps the Repeat exclusively, DD means that both ends of the
  fragment overlap both (dual) exon and repeat ...

\item[Chromosome / EStart / EEnd / EStrand] Start, end and strand of the exon

\item[RStart / REnd / RStrand] Start, end and strand of the repeat

\item[RepeatRank] Relative exon/intron position of the repeat to the contig

\item[UpExonStart / UpExonEnd] Coordinates used for calculating expression of
  genome immediatly adjacent an exon boundary. Useful for quantifying read-
  through or spurious transcriptional events.

\item[UpThread] The number of read 'threads' going upstream of the exon.
  See Manuscript for a figure explaining this.

\item[DownThread] The number of read 'threads' going downstream of the exon.
  See Manuscript for a figure explaining this.

\item[ExonRPKM] RPKM calculation for this exon

\item[ExonMax] The maximum coverage count reached within the exon boundaries. Often
  more reliable measure of expression then RPKM for small exons.

\item[UpExonRPKM / UpExonMax] The expression of the exon immediatly upstream of the
  one this row is referring to. (i.e. Exon 1 expression if the row refers
  to Exon 2). Useful for quantifying the relative increase in expression
  when a TE is acting as an alternative promoter into a downstream exon.

\item[RepeatRPKM / RepeatMaxCoverage]  Expression level within repeat boundaries.

\item[UpstreamRepeatRPKM / UpstreamRepeatMaxCoverage] The expression adjacent to the
  repeat, a test for background expression levels.

\item[RefID] When intersecting to a reference gene set, the gene symbol of any genes
  which intersect the area between the Exon-Repeat coordinates.

\item[RefStrand] Strand of the reference genes defined above

\item[assXref] The strand-relationship between the reference gene and the contig exon
  This accounts for anti-sense long non-coding RNA (as), or transcripts
  which run anti-sense to the reference gene. (s) is sense and (c) means
  complex, often some combination of multiple genes. (u) means it could not
  be determined.

\item[Contribution] An estimate of the promoter contribution of this Repeat TSS to
  the expression of gene in total. Calculated with ExonMax and UpExonMax.

\item[UpCov] Ratio of the coverage adjacent to an exon and the exon expression

\item[UpExonRatio] Ratio of hte expression of the exon and it's upstream exon

\item[ThreadRatio] DownThread / UpThread. Set to [10] if dividing by zero.

\item[RepeatID] A unique Identifer for each Repeat in the genome (left-most
  coordinate). Can repeat and thus be used for determining one repeat
  inititating a trancsript in different assemblies.

\item[LIBRARY] Library from which this repeat-exon interaction was calculated from.
\end{description}


\subsubsection{`\arrows{project}.rslions'}
\begin{description}
\item[Normal\_occ] Number of times each TE-initiation was found in the ''normal'' set of libraries. (Usually set 1)

\item[Cancer\_occ] Number of times each TE-initiation was found in the ''cancer'' set of libraries. (Usually set 2)

\item[Library] A semi-colon seperated list of the LIBRARY identifiers in which each TE-initiation was found in.
\end{description}  

\clearpage
\appendix
%\begin{appendices}
\section{Appendices}
\subsection{Running Docker on the command line}

Often it is useful to detach the running Docker container from the terminal; this can be acheived in a number of ways. The official Docker method is to detach the terminal using 

\texttt{Ctrl-p Ctrl-q}

to disconnect, however this is dependent on using the \texttt{\textbf{-t}} option when running the container. Once detached the user can reconnect to the container using

\begin{bash}
docker attach <container-id or name >
\end{bash}

It is possible to name a docker container using the \texttt{-n <name>} option to the \bashinline{docker run} command. However where the container name is not specified or known,
the container-id can be found using the \bashinline{docker ps} command.

\vspace{2em}

The author prefers using GNU Screen as it is capable of presenting a split screen allowing the user to monitor LIONS whilst continuing to work. 
Screen is invoked using the command \bashinline{screen} which opens a virtual terminal, the LIONS docker can then be run as described in section \ref{lions_run}. 

Screen sessions can be detached using \texttt{Ctrl-a d} and reconnected using \bashinline{screen -r}. 

A single screen session allows a user to simultaneously run multiple virtual terminals; known as `windows'. New windows are created within a session using \texttt{Ctrl-a c}, whilst \texttt{Ctrl-a n} and \texttt{Ctrl-a p} allow the user to switch between windows within a session.

Screen is able to split the terminal view vertically using \texttt{Ctrl-a |} or \texttt{Ctrl-a V}, depending on the implementation; \texttt{Ctrl-a S} can be used to split the window horizontally. Once split the user can switch between view regions using \texttt{Ctrl-a Tab}.


%\end{appendices}
\end{document}
